### ISSUE TITLE
Registry returned 400 for DELETE of one of my packages

### ISSUE THREAD

**Issue by ravicious** [npm/npm#16684]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

When I try to unpublish my package, [`@ravicious/react-text-mask`](https://www.npmjs.com/package/@ravicious/react-text-mask), I get the following error:

<details>

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/Cellar/node/7.10.0/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'unpublish',
1 verbose cli   '--force',
1 verbose cli   '@ravicious/react-text-mask@5.0.3' ]
2 info using npm@4.2.0
3 info using node@v7.10.0
4 verbose config Skipping project config: /Users/rav/.npmrc. (matches userconfig)
5 warn using --force I sure hope you know what you are doing.
6 silly unpublish args[0] @ravicious/react-text-mask@5.0.3
7 silly unpublish thing Result {
7 silly unpublish   raw: '@ravicious/react-text-mask@5.0.3',
7 silly unpublish   scope: '@ravicious',
7 silly unpublish   escapedName: '@ravicious%2freact-text-mask',
7 silly unpublish   name: '@ravicious/react-text-mask',
7 silly unpublish   rawSpec: '5.0.3',
7 silly unpublish   spec: '5.0.3',
7 silly unpublish   type: 'version' }
8 verbose getPublishConfig null
9 silly ls normalized @ravicious/react-text-mask/5.0.3
10 silly gentlyRm /Users/rav/.npm/@ravicious/react-text-mask/5.0.3 is being purged
11 verbose gentlyRm don't care about contents; nuking /Users/rav/.npm/@ravicious/react-text-mask/5.0.3
12 silly mapToRegistry name @ravicious/react-text-mask
13 silly mapToRegistry scope (from package name) @ravicious
14 verbose mapToRegistry no registry URL found in name for scope @ravicious
15 silly mapToRegistry using default registry
16 silly mapToRegistry registry https://registry.npmjs.org/
17 silly mapToRegistry data Result {
17 silly mapToRegistry   raw: '@ravicious/react-text-mask',
17 silly mapToRegistry   scope: '@ravicious',
17 silly mapToRegistry   escapedName: '@ravicious%2freact-text-mask',
17 silly mapToRegistry   name: '@ravicious/react-text-mask',
17 silly mapToRegistry   rawSpec: '',
17 silly mapToRegistry   spec: 'latest',
17 silly mapToRegistry   type: 'tag' }
18 silly mapToRegistry uri https://registry.npmjs.org/@ravicious%2freact-text-mask
19 verbose get GET as part of write; not caching result
20 verbose request uri https://registry.npmjs.org/@ravicious%2freact-text-mask?write=true
21 verbose request no auth needed
22 info attempt registry request try #1 at 10:05:10 PM
23 verbose request using bearer token for auth
24 verbose request id 97f21c3923f81f98
25 http request GET https://registry.npmjs.org/@ravicious%2freact-text-mask?write=true
26 http 200 https://registry.npmjs.org/@ravicious%2freact-text-mask?write=true
27 verbose headers { server: 'CouchDB/1.5.0 (Erlang OTP/R16B03)',
27 verbose headers   'content-type': 'application/json',
27 verbose headers   etag: '"59234462-fcf"',
27 verbose headers   'content-encoding': 'gzip',
27 verbose headers   'cache-control': 'max-age=0',
27 verbose headers   'transfer-encoding': 'chunked',
27 verbose headers   'accept-ranges': 'bytes',
27 verbose headers   date: 'Mon, 22 May 2017 20:05:11 GMT',
27 verbose headers   via: '1.1 varnish',
27 verbose headers   connection: 'keep-alive',
27 verbose headers   'x-served-by': 'cache-ams4147-AMS',
27 verbose headers   'x-cache': 'MISS',
27 verbose headers   'x-cache-hits': '0',
27 verbose headers   'x-timer': 'S1495483510.483778,VS0,VE605',
27 verbose headers   vary: 'Accept-Encoding' }
28 silly get cb [ 200,
28 silly get   { server: 'CouchDB/1.5.0 (Erlang OTP/R16B03)',
28 silly get     'content-type': 'application/json',
28 silly get     etag: '"59234462-fcf"',
28 silly get     'content-encoding': 'gzip',
28 silly get     'cache-control': 'max-age=0',
28 silly get     'transfer-encoding': 'chunked',
28 silly get     'accept-ranges': 'bytes',
28 silly get     date: 'Mon, 22 May 2017 20:05:11 GMT',
28 silly get     via: '1.1 varnish',
28 silly get     connection: 'keep-alive',
28 silly get     'x-served-by': 'cache-ams4147-AMS',
28 silly get     'x-cache': 'MISS',
28 silly get     'x-cache-hits': '0',
28 silly get     'x-timer': 'S1495483510.483778,VS0,VE605',
28 silly get     vary: 'Accept-Encoding' } ]
29 verbose get saving @ravicious/react-text-mask to /Users/rav/.npm/registry.npmjs.org/_40ravicious_252freact-text-mask_3Fwrite_3Dtrue/.cache.json
30 verbose correctMkdir /Users/rav/.npm correctMkdir not in flight; initializing
31 verbose unpublish removing attachments { shasum: 'a952ceb47cd388a6383c3aa9a7e5e3a99aae4568',
31 verbose unpublish   tarball: 'https://registry.npmjs.org/@ravicious/react-text-mask/-/react-text-mask-5.0.3.tgz' }
32 info unpublish No versions remain, removing entire package
33 verbose request uri https://registry.npmjs.org/@ravicious%2freact-text-mask/-rev/7-025632ded0cf63f229946aad35da16a1
34 verbose request sending authorization for write operation
35 info attempt registry request try #1 at 10:05:11 PM
36 verbose request using bearer token for auth
37 http request DELETE https://registry.npmjs.org/@ravicious%2freact-text-mask/-rev/7-025632ded0cf63f229946aad35da16a1
38 http 400 https://registry.npmjs.org/@ravicious%2freact-text-mask/-rev/7-025632ded0cf63f229946aad35da16a1
39 verbose headers { 'content-type': 'application/json',
39 verbose headers   'cache-control': 'max-age=0',
39 verbose headers   'content-length': '17',
39 verbose headers   'accept-ranges': 'bytes',
39 verbose headers   date: 'Mon, 22 May 2017 20:05:11 GMT',
39 verbose headers   via: '1.1 varnish',
39 verbose headers   connection: 'keep-alive',
39 verbose headers   'x-served-by': 'cache-ams4147-AMS',
39 verbose headers   'x-cache': 'MISS',
39 verbose headers   'x-cache-hits': '0',
39 verbose headers   'x-timer': 'S1495483511.142639,VS0,VE824' }
40 verbose request invalidating /Users/rav/.npm/registry.npmjs.org/_40ravicious_252freact-text-mask on DELETE
41 verbose stack Error: Registry returned 400 for DELETE on https://registry.npmjs.org/@ravicious%2freact-text-mask/-rev/7-025632ded0cf63f229946aad35da16a1
41 verbose stack     at makeError (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:304:12)
41 verbose stack     at CachingRegistryClient.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:282:14)
41 verbose stack     at Request._callback (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:212:14)
41 verbose stack     at Request.self.callback (/usr/local/lib/node_modules/npm/node_modules/request/request.js:186:22)
41 verbose stack     at emitTwo (events.js:106:13)
41 verbose stack     at Request.emit (events.js:194:7)
41 verbose stack     at Request.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1081:10)
41 verbose stack     at emitOne (events.js:96:13)
41 verbose stack     at Request.emit (events.js:191:7)
41 verbose stack     at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1001:12)
42 verbose statusCode 400
43 verbose pkgid @ravicious/react-text-mask
44 verbose cwd /Users/rav
45 error Darwin 16.6.0
46 error argv "/usr/local/Cellar/node/7.10.0/bin/node" "/usr/local/bin/npm" "unpublish" "--force" "@ravicious/react-text-mask@5.0.3"
47 error node v7.10.0
48 error npm  v4.2.0
49 error code E400
50 error Registry returned 400 for DELETE on https://registry.npmjs.org/@ravicious%2freact-text-mask/-rev/7-025632ded0cf63f229946aad35da16a1
51 error If you need help, you may report this error at:
51 error     <https://github.com/npm/npm/issues>
52 verbose exit [ 1, true ]
```
</details>

#### How can the CLI team reproduce the problem?

Try to remove the package I guess. 🤔

The debug log is above under details.

### supporting information:

 - `npm -v` prints: 4.2.0
 - `node -v` prints: 7.10.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by MarcelRobitaille** [comment]
Did you ever figure this out?

**Comment by ravicious** [comment]
Yeah, I got 400 because I tried to unpublish a package that was public for over 24 hours. I wish the error message was more clear. AFAIR somebody already made an issue with a suggestion to improve that.

> With the default registry (registry.npmjs.org), unpublish is only allowed with versions published in the last 24 hours. If you are trying to unpublish a version published longer ago than that, contact support@npmjs.com.

https://docs.npmjs.com/cli/unpublish

In my case, they unpublished my package rather fast, as it was a scoped package and no one was really downloading it.
